src/file_uploader/s3.rs

```diff
@@ -0,0 +1,89 @@
+use crate::{handlers::upload::UploadError, util::load_env};
+use aws_sdk_s3::Client;
+use aws_sdk_s3::{
+    config::{BehaviorVersion, Region},
+    primitives::ByteStream,
+};
+use std::{env, fs::File, io::Read, path::Path};
+
+pub trait S3Client: Sized {
+    fn new() -> impl std::future::Future<Output = Result<Self, UploadError>> + Send;
+    fn upload_file_to_s3(
+        &self,
+        path: &Path,
+        file_name: String,
+    ) -> impl std::future::Future<Output = Result<(), UploadError>> + Send;
```

Comment:
Interesting choice for explicitly using future response type instead of prefixing function with `async`; curious to know if there's a reason?
No issue with implementation; just curious 😅

Reply:
I assumed it was because `async` in public traits are still problematic. https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html#async-fn-in-public-traits

Reply:
That's right. Here's the full warning
![image](https://github.com/user-attachments/assets/acc78258-785e-421a-b94b-3a8144b0b7ff)